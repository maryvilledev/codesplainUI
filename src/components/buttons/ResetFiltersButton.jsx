import React, { PropTypes } from 'react';
import { RaisedButton } from 'material-ui';

const ResetFiltersButton = ({resetCallback}) => (
  <RaisedButton
    label="Reset Filters"
    onTouchTap={resetCallback}
    secondary
  />
);

ResetFiltersButton.propTypes = {
  resetCallback: PropTypes.func.isRequired,
};

export default ResetFiltersButton;
