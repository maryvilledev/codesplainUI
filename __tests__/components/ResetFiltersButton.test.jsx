import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import ResetFiltersButton from '../../src/components/buttons/ResetFiltersButton';

describe('<ResetFiltersButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <ResetFiltersButton
        resetCallback={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: resetCallback', () => {
    it('is triggered when button is clicked', () => {
      const resetCallback = jest.fn();
      const wrapper = shallowWithContext(
        <ResetFiltersButton
          resetCallback={resetCallback}
        />
      );
      const button = wrapper.find('RaisedButton');
      button.simulate('touchTap');
      expect(resetCallback).toHaveBeenCalled();
    });
  });
});
