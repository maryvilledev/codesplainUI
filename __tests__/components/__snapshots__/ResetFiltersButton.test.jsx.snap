exports[`<ResetFiltersButton /> matches snapshot 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={true}
  label="Reset"
  labelPosition="after"
  labelStyle={
    Object {
      "fontSize": "12px",
    }
  }
  onTouchTap={[Function]}
  primary={false}
  secondary={true} />
`;
